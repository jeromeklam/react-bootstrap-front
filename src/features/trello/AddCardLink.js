import React from 'react';
import PropTypes from 'prop-types';

export default function AddCardLink(props) {
  return (
    <button className="btn btn-primary" onClick={props.onClick}>
      {props.t({ id: 'rbf.trello.card.addLink', defaultMessage: 'Click to add card' })}
    </button>
  );
}

AddCardLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

AddCardLink.defaultProps = {};
