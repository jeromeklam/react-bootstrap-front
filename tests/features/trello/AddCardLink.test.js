import React from 'react';
import renderer from 'react-test-renderer';
import AddCardLink from '../../../src/features/trello/AddCardLink';

const callback = jest.fn();
jest.useFakeTimers();

it('renders standard', () => {
  // First render
  const component = renderer.create(
    <AddCardLink onClick={callback} t={callback} />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
