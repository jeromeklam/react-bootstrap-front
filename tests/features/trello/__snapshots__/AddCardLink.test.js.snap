// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders standard 1`] = `
<button
  className="btn btn-primary"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "defaultMessage": "Click to add card",
            "id": "rbf.trello.card.addLink",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;
